scene/gui/rich_text_label.cpp

```diff
@@ -3929,6 +3930,11 @@ void RichTextLabel::_bind_methods() {
 
 void RichTextLabel::set_visible_characters(int p_visible) {
 	visible_characters = p_visible;
+	int total_char_count = get_total_character_count();
+	if (total_char_count > 0) {
```

Comment:
Need to set `percent_visible = 1` where `p_visible == -1` else `percent_visible` will be negative.

Reply:
Hey, thanks for the feedback! I've added a check to handle the corner case.